zipkin-server/src/main/java/zipkin2/server/internal/kafka/ZipkinKafkaCollectorConfiguration.java

```diff
@@ -33,15 +39,27 @@
 @Conditional(ZipkinKafkaCollectorConfiguration.KafkaBootstrapServersSet.class)
 @EnableConfigurationProperties(ZipkinKafkaCollectorProperties.class)
 public class ZipkinKafkaCollectorConfiguration { // makes simple type name unique for /actuator/conditions
+  static final String QUALIFIER = "zipkinKafka";
 
-  @Bean(initMethod = "start")
-  KafkaCollector kafka(
+  @Bean(initMethod = "start") @ConditionalOnSelfTracing KafkaCollector kafka(
       ZipkinKafkaCollectorProperties properties,
       CollectorSampler sampler,
       CollectorMetrics metrics,
-      StorageComponent storage) {
-    return properties.toBuilder().sampler(sampler).metrics(metrics).storage(storage).build();
+      StorageComponent storage,
+      Optional<Tracing> maybeTracing) {
+    final KafkaCollector.Builder builder = properties.toBuilder()
+      .sampler(sampler)
+      .metrics(metrics)
+      .storage(storage);
+    if (maybeTracing.isPresent()) {
+      final KafkaTracing kafkaTracing = KafkaTracing.newBuilder(maybeTracing.get())
+        .remoteServiceName("zipkin-kafka")
+        .build();
+      builder.consumerSupplier(props -> kafkaTracing.consumer(new KafkaConsumer<>(props)));
```

Comment:
maybe add another inject parameter for consumerSupplier instead? you can then make a bean that is conditionally wrapped with KafkaTracing

Reply:
gave a try, let me know how it looks now